pkgs/applications/virtualization/singularity/template.nix

```diff
@@ -0,0 +1,163 @@
+{ lib
+, util-linux
+, gpgme
+, openssl
+, libuuid
+, bash
+, coreutils
+, which
+, makeWrapper
+, cryptsetup
+, squashfsTools
+, nvidia-docker # For nvidia-container-cli
+, glibc # For ldconfig needed by nvidia-container-cli
+, go
+, buildGoModule
+, pname
+, version
+, src
+, projectName # "apptainer" or "singularity"
+  # These cannot be correctly override with `overrideAttrs`
+, vendorSha256 ? null
+, deleteVendor ? false
+, proxyVendor ? false
+, doCheck ? true
+, withoutSuid ? true # Required to run containers without suid
+, enableNvidiaContainerCli ? true
+}:
+
+with lib;
+let
+  propagatedBuildInputs = [ coreutils squashfsTools bash ];
+in
+buildGoModule {
+  inherit pname version src;
+
+  # `buildGoModule` reads the `go.mod` file,
+  # so no need to specify `goPackagePath`
+
+  # All three projects provide vendored source tarball
+  # so we set vendorSha256 = null;
+  # One who use unvendored source can override this
+  # with `override`
+  inherit vendorSha256 deleteVendor proxyVendor;
+
+  # apptainer and singularity-ce both want to reference Go
+  # to compile the extensions when building container images
+  allowGoReference = true;
+
+  passthru = {
+    inherit projectName;
+  };
+
+  buildInputs = [ gpgme openssl libuuid ];
+  nativeBuildInputs = [ util-linux which makeWrapper cryptsetup ];
+  inherit propagatedBuildInputs;
+
+  outputs = [ "out" "man" ];
+
+  postPatch = ''
+    substituteInPlace internal/pkg/build/files/copy.go \
+      --replace /bin/cp ${coreutils}/bin/cp
+  '';
+
+  configureScript = "./mconfig";
+
+  configureFlags = [
+    "--localstatedir=/var"
+  ]
+  ++ lib.optional withoutSuid "--without-suid";
+
+  # Using `buildGoModule` instead of `buildGoPackage`,
+  # We no longer need to `cd go/src/${goPackagePath}`
```

Comment:
I guess out of the context of previous revisions this comment isn't going to be of much help anyway:)